src/analyser.cpp

```diff
@@ -278,6 +402,14 @@ bool AnalyserInternalEquation::check(const AnalyserModelPtr &analyserModel, bool
                                    mVariables.front() :
                                    nullptr;
 
+    // If we have one variable left, but it's not isolated, try to rearrange it
+    if ((unknownVariableLeft != nullptr) && !variableOnLhsOrRhs(unknownVariableLeft)) {
+        auto newAst = rearrange(unknownVariableLeft);
+        if (newAst != nullptr) {
+            mAst = newAst;
+        }
+    }
```

Comment:
Remember that you will also need to update the type of some variables and of the equation.